### PR TITLE
feat: add Agent

### PR DESCRIPTION
**Pull request by Laura-Berkers**

### CODE REVIEW

File: Aplib.Core/Agent.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+{
+    /// <summary>
+    /// Represents an agent that performs actions based on goals and beliefs.
+    /// </summary>
+    public class Agent : IAgent
+    {
+        /// <summary>
+        /// Gets the beliefset of the agent.
+        /// </summary>
+        public Beliefset State { get; }
```
Comment (JensSteenmetz) [inline Aplib.Core/Agent.cs]: More descriptive name:
```suggestion
        public Beliefset Beliefset { get; }
```

File: Aplib.Core/Agent.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        /// <summary>
+        /// Gets the goal structure of the agent.
+        /// </summary>
+        public GoalStructure Goals { get; }
```
Comment (JensSteenmetz) [inline Aplib.Core/Agent.cs]: Is it correct that we're not using the Desireset stub right now?

**Review by JensSteenmetz — CHANGES_REQUESTED** [review]
Looks good overall, some small naming suggestions.

File: Aplib.Core/Agent.cs (hunk 2 above)
Comment (joachimdekker) [reply Aplib.Core/Agent.cs]: You're right. I'll fix that once we've implemented that!

File: Aplib.Core/BdiAgent.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+            // Desire
+            _desireSet.UpdateStatus(_beliefSet);
+            if (Status != CompletionStatus.Unfinished)
+                return;
+            IGoal goal = _desireSet.GetCurrentGoal(_beliefSet);
+
+
+            // Intent
+            Tactic tactic = goal.Tactic;
```
Comment (JensSteenmetz) [inline Aplib.Core/BdiAgent.cs]:
```suggestion
            IGoal goal = _desireSet.GetCurrentGoal(_beliefSet);

            // Intent
            Tactic tactic = goal.Tactic;
```

**Review by JensSteenmetz — CHANGES_REQUESTED** [review]
Nice work! One small nitpick

**Review by JensSteenmetz — APPROVED** [review]
LGTM

**Review by joachimdekker — APPROVED** [review]
This is a test.

**Pull request merged**